frontend/src/routes/manage/Realm/DangerZone.tsx

```diff
@@ -209,25 +204,21 @@ const RemoveRealm: React.FC<InnerProps> = ({ realm }) => {
             {t("manage.realm.danger-zone.delete.warning")}
         </p>
         <div css={{ marginTop: 32, textAlign: "center" }}>
-            <Button kind="danger" onClick={() => setModalActive(true)}>
+            <Button kind="danger" onClick={() => modalRef.current?.open()}>
                 <span>{buttonContent}</span>
             </Button>
         </div>
-        {modalActive && (
-            <Modal title={t("manage.are-you-sure")} close={() => setModalActive(false)}>
-                <p>
-                    <Trans i18nKey="manage.realm.danger-zone.delete.cannot-be-undone">
-                        foo<strong>bar</strong>baz
-                    </Trans>
-                </p>
-                <form onSubmit={remove} css={{ marginTop: 32, textAlign: "center" }}>
-                    <Button kind="danger">
-                        <span>{buttonContent}</span>
-                    </Button>
-                    {isInFlight && <div css={{ marginTop: 16 }}><Spinner size={20} /></div>}
-                </form>
-                {boxError(commitError)}
-            </Modal>
-        )}
+        <ConfirmationModal
+            title={t("manage.are-you-sure")}
+            {...{ buttonContent }}
+            onSubmit={remove}
+            ref={modalRef}
```

Comment:
I roughly order fields by "relevance" regardless of how I pass them. If you insist on this style, I think I would rather get rid of the spread.

Note also that this can never be a hard and fast rule. There is a semantic difference. If props are shared between the key-value pairs and the spread, the order determines which one wins. And I can imagine use cases for both to win. (Basically: Are the attributes just a set of defaults overridable with the user-passed spread or do you want to explicitly prevent the spread overriding some props.)

Reply:
Ah ye ok. Then let's leave it like that.